packages/portal-contracts/src/Portal.sol

```diff
@@ -434,4 +457,23 @@ contract Portal is Owned {
         bool success = token.transfer(msg.sender, tok);
         require(success, "transfer failed");
     }
+
+    function openEscrowKey(bytes20 scriptHash, uint256 amountSats) public view returns (bytes32) {
+	    return keccak256(abi.encode(scriptHash, amountSats));
+    }
+
+    function addOpenEscrow(bytes20 scriptHash, uint256 amountSats) private {
+	bytes32 recKey = openEscrowKey(scriptHash, amountSats);
+	uint256 existingOpenEscrow = openEscrows[recKey];
+	require(existingOpenEscrow == 0, "Escrow collision, please retry");
+	// Say Alice opens an escrow at block height 1000. She submits a Bitcoin transaction.
+	// A normal two-block reorg occurs, and her transaction ends up confirmed at block height 999.
+	openEscrows[recKey] = btcVerifier.mirror().getLatestBlockHeight() - minConfirmations;
```

Comment:
Has to be `- minConfirmations + 1`, I think.

Otherwise, say block 1000 just came out. Alice proves that she sent Bitcoin in block 995 = 5 confirmations = closes escrow.

Bob trades against the same order, reuses her proof.

The +1 ensures Bob's proof has to be in block 996 or later.

Reply:
The check condition is exclusive, so must be > than this number, which I think obviates the need for the +1 here